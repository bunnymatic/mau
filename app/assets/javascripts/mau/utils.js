// Generated by CoffeeScript 1.12.7
(function() {
  var MAU;

  MAU = window.MAU = window.MAU || {};

  MAU.Utils = {
    ellipsize: function(str, max, ellipse, chars, truncate) {
      var DEFAULTS, c, i, last, len, ref;
      DEFAULTS = {
        ellipse: 'â€¦',
        chars: [' ', '-'],
        max: 140,
        truncate: true
      };
      max || (max = DEFAULTS.max);
      chars || (chars = DEFAULTS.chars);
      ellipse || (ellipse = DEFAULTS.ellipse);
      truncate || (truncate = DEFAULTS.truncate);
      last = 0;
      c = '';
      if (str.length < max) {
        return str;
      }
      i = 0;
      len = str.length;
      while (i < len) {
        c = str.charAt(i);
        i++;
        if (chars.indexOf(c) !== -1) {
          last = i;
          if (i < max) {
            continue;
          }
          if (last === 0) {
            return (ref = !truncate) != null ? ref : {
              '': str.substring(0, max - 1) + ellipse
            };
          }
          return str.substring(0, last) + ellipse;
        }
      }
      return str;
    },
    debounce: function(func, threshold, execAsap) {
      var timeout;
      timeout = null;
      return function() {
        var args, delayed, obj;
        obj = this;
        args = arguments;
        delayed = function() {
          if (!execAsap) {
            func.apply(obj, args);
          }
          return timeout = null;
        };
        if (timeout) {
          clearTimeout(timeout);
        } else if (execAsap) {
          func.apply(obj, args);
        }
        return timeout = setTimeout(delayed, threshold || 100);
      };
    },
    validateEmail: function(str) {
      return (str.indexOf(".") > 2) && (str.indexOf("@") > 0);
    },
    getSize: function(el) {
      return {
        width: el.offsetWidth,
        height: el.offsetHeight
      };
    },
    getPosition: function(el) {
      var xx, yy;
      xx = yy = 0;
      while (el) {
        xx += element.offsetLeft - element.scrollLeft + element.clientLeft;
        xx += element.offsetTop - element.scrollTop + element.clientTop;
        el = element.offsetParent;
      }
      return {
        x: xx,
        y: yy
      };
    },
    createElement: function(tagName, attrs) {
      el;
      var el, k, v;
      if (document && document.createElement) {
        el = document.createElement(tagName);
        for (k in attrs) {
          v = attrs[k];
          el.setAttribute(k, v);
        }
      }
      return el;
    },
    post_to_url: function(path, params, method) {
      var form, hiddenField, key, val;
      method = method || "post";
      form = this.createElement('form', {
        method: method,
        action: path
      });
      hiddenField = null;
      for (key in params) {
        val = params[key];
        hiddenField = this.createElement('input', {
          type: 'hidden',
          name: key,
          value: val
        });
        form.appendChild(hiddenField);
      }
      hiddenField = this.createElement('input', {
        type: 'hidden',
        name: 'authenticity_token',
        value: unescape(authenticityToken)
      });
      form.appendChild(hiddenField);
      document.body.appendChild(form);
      return form.submit();
    }
  };

}).call(this);
