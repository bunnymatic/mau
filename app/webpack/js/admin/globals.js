/**
 * Here we import things we want to be global (like jQuery plugins or similar classes)
 * So they can be accessed from views in a script tag
 *
 * An easy way to make them global is simply to stick them on window.
 **/

import MauTimeago from "../app/mau_timeago";
import AdminRoleManager from "./admin_role_manager";
import ApplicationEventsDetailsForm from "./application_events_details_form";
import ArtistsAdminIndex from "./artists_admin_index";
import { GraphPerDay, PlainGraph } from "./graph_helpers";
import MarkItDown from "./mark_it_down";
import MauDatatables from "./mau_datatables";
import SlideToggler from "./slide_toggler";
import StudioArranger from "./studio_arranger";

const globals = {
  AdminRoleManager,
  ArtistsAdminIndex,
  ApplicationEventsDetailsForm,
  GraphPerDay,
  MarkItDown,
  MauDatatables,
  MauTimeago,
  PlainGraph,
  SlideToggler,
  StudioArranger,
};

window.MAU = globals;
