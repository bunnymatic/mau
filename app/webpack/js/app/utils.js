const debounce = function (func, threshold, execAsap) {
  var timeout;
  timeout = null;
  return function () {
    var args, delayed, obj;
    obj = this;
    args = arguments;
    delayed = function () {
      if (!execAsap) {
        func.apply(obj, args);
      }
      return (timeout = null);
    };
    if (timeout) {
      clearTimeout(timeout);
    } else if (execAsap) {
      func.apply(obj, args);
    }
    return (timeout = setTimeout(delayed, threshold || 100));
  };
};

export { debounce };
