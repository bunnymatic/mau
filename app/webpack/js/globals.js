// go with maps
import "core-js/stable";
import "regenerator-runtime/runtime";

import AccordionShowSectionGivenHash from "./app/accordion_show_section_given_hash";
import AccordionShowSectionWithErrors from "./app/accordion_show_section_with_errors";
import ArrangeArtForm from "./app/arrange_art_form";
import ArtPieceForm from "./app/art_piece_form";
import ArtistBioToggler from "./app/artist_bio_toggler";
import ArtistDetailsToggler from "./app/artist_details_toggler";
import ArtistListInfiniteScroller from "./app/artist_list_infinite_scroller";
import BackToTop from "./app/back_to_top";
import MapResizer from "./app/map_resizer";
import MauMap from "./app/mau_map";
import MobileNavigation from "./app/mobile_navigation";
import Navigation from "./app/navigation";
import PasswordStrengthMeter from "./app/password_strength_meter";
import Sampler from "./app/sampler";
import WhatsThisPopup from "./app/whats_this_popup";

const globals = {
  AccordionShowSectionGivenHash,
  AccordionShowSectionWithErrors,
  ArrangeArtForm,
  ArtistBioToggler,
  ArtistDetailsToggler,
  ArtistListInfiniteScroller,
  ArtPieceForm,
  BackToTop,
  MauMap,
  MapResizer,
  MobileNavigation,
  Navigation,
  PasswordStrengthMeter,
  Sampler,
  WhatsThisPopup,
};
window.MAU = globals;
