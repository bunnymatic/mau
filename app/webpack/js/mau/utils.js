const ellipsize = function(str, max, ellipse, chars, truncate) {
  var DEFAULTS, c, i, last, len, ref;
  DEFAULTS = {
    ellipse: "â€¦",
    chars: [" ", "-"],
    max: 140,
    truncate: true
  };
  max || (max = DEFAULTS.max);
  chars || (chars = DEFAULTS.chars);
  ellipse || (ellipse = DEFAULTS.ellipse);
  truncate || (truncate = DEFAULTS.truncate);
  last = 0;
  c = "";
  if (str.length < max) {
    return str;
  }
  i = 0;
  len = str.length;
  while (i < len) {
    c = str.charAt(i);
    i++;
    if (chars.indexOf(c) !== -1) {
      last = i;
      if (i < max) {
        continue;
      }
      if (last === 0) {
        return (ref = !truncate) != null
          ? ref
          : {
              "": str.substring(0, max - 1) + ellipse
            };
      }
      return str.substring(0, last) + ellipse;
    }
  }
  return str;
};
const debounce = function(func, threshold, execAsap) {
  var timeout;
  timeout = null;
  return function() {
    var args, delayed, obj;
    obj = this;
    args = arguments;
    delayed = function() {
      if (!execAsap) {
        func.apply(obj, args);
      }
      return (timeout = null);
    };
    if (timeout) {
      clearTimeout(timeout);
    } else if (execAsap) {
      func.apply(obj, args);
    }
    return (timeout = setTimeout(delayed, threshold || 100));
  };
};

export { debounce, ellipsize };
