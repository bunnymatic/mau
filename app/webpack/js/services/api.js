import { destroy, get, post } from "@js/services/mau_ajax";
import { camelizeKeys } from "humps";

const camelize = (data) => camelizeKeys(data);

export const api = {
  applicationEvents: {
    index: ({ since }) =>
      get("/admin/application_events.json", { "query[since]": since }).then(
        camelize
      ),
  },
  emailLists: {
    emails: {
      index: (emailListId) =>
        get(`/admin/email_lists/${emailListId}/emails`).then(camelize),
      save: (emailListId, data) =>
        post(`/admin/email_lists/${emailListId}/emails`, data).then(camelize),
      remove: (id, emailListId) =>
        destroy(`/admin/email_lists/${emailListId}/emails/${id}`).then(
          camelize
        ),
    },
  },
  favorites: {
    add: (userSlug, type, id) => {
      const url = `/users/${userSlug}/favorites`;
      return post(url, { favorite: { type, id } });
    },
  },

  notes: {
    create: (data) => post("/api/notes", data),
  },
  search: {
    query: (queryParams) => {
      return post("/search.json", queryParams);
    },
  },
  users: {
    registerForOs: (artistSlug, data) => {
      const url = `/api/artists/${artistSlug}/register_for_open_studios`;
      return post(url, data).then(camelize);
    },
    whoami: () => get("/users/whoami").then(camelize),
  },
};
