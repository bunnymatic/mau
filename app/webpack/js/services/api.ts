import * as types from "@reactjs/types";
import { destroy, get, post, put } from "@services/mau_ajax";
import { camelizeKeys } from "humps";

const camelize = (data) => camelizeKeys(data);

export const api = {
  openStudios: {
    submitRegistrationStatus: (status: boolean) => {
      return api.users.whoami().then(function ({ currentUser }) {
        if (currentUser && currentUser.slug) {
          return api.users.registerForOs(currentUser.slug, status);
        }
      });
    },
    participants: {
      update: ({
        id,
        artistId,
        openStudiosParticipant,
      }: types.OpenStudiosParticipantUpdateRequest) => {
        return put(`/api/artists/${artistId}/open_studios_participants/${id}`, {
          openStudiosParticipant,
        }).then(camelize);
      },
    },
  },
  artPieces: {
    contact: (id: types.IdType, formData: types.ContactArtistFormData) => {
      return post(`/api/v2/art_pieces/${id}/contact`, formData).then(camelize);
    },
  },
  applicationEvents: {
    index: ({ since }): Promise<types.ApplicationEventsListResponse> =>
      get("/admin/application_events.json", { "query[since]": since }).then(
        camelize
      ),
  },
  emailLists: {
    emails: {
      index: (emailListId) =>
        get(`/admin/email_lists/${emailListId}/emails`).then(camelize),
      save: (emailListId, data) =>
        post(`/admin/email_lists/${emailListId}/emails`, data).then(camelize),
      remove: (id, emailListId) =>
        destroy(`/admin/email_lists/${emailListId}/emails/${id}`).then(
          camelize
        ),
    },
  },
  favorites: {
    add: (userSlug, type, id) => {
      const url = `/users/${userSlug}/favorites`;
      return post(url, { favorite: { type, id } });
    },
  },

  notes: {
    create: (data) => post("/api/notes", data),
  },
  search: {
    query: (queryParams) => {
      return post("/search.json", queryParams);
    },
  },
  users: {
    registerForOs: (artistSlug, status) => {
      const url = `/api/artists/${artistSlug}/register_for_open_studios`;
      return post(url, { participation: status }).then(camelize);
    },
    whoami: () => get("/users/whoami").then(camelize),
  },
};
