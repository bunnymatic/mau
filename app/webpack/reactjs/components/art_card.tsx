import { ArtPiece } from "@models/art_piece.model";
import { ArtModal } from "@reactjs/components/art_modal";
import cx from "classnames";
import React, { FC } from "react";

interface AttributionProps {
  artPiece: ArtPiece;
}

const AttributionItems: FC<AttributionProps> = ({ artPiece }) => {
  if (!artPiece.dimensions && !artPiece.price) {
    return <div className="art-card__description-item">{artPiece.title}</div>;
  }
  return (
    <>
      <div className="art-card__description-item">
        {Boolean(artPiece.dimensions) && artPiece.dimensions}
      </div>
      <div className="art-card__description-item">
        {Boolean(artPiece.price) && (
          <span
            className={cx("art-card__price", {
              "art-card__price--sold": artPiece.hasSold(),
            })}
          >
            {artPiece.displayPrice}
          </span>
        )}
      </div>
    </>
  );
};

interface ArtCardProps {
  artPiece: ArtPiece;
  classes?: string;
}

export const ArtCard: FC<ArtCardProps> = ({ artPiece, classes }) => {
  return (
    <div className={cx("art-card", classes)}>
      <ArtModal artPiece={artPiece}>
        <div
          className="image"
          style={{ backgroundImage: `url("${artPiece.imageUrls.original}")` }}
        ></div>
        {Boolean(artPiece.hasSold()) && (
          <span className="art-card__sold">Sold</span>
        )}
        <div className="art-card__description">
          <AttributionItems artPiece={artPiece} />
        </div>
      </ArtModal>
    </div>
  );
};
