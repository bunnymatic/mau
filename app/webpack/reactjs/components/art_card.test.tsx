import { describe, expect, it } from "@jest/globals";
import { ArtPiece } from "@models/art_piece.model";
import { jsonApiArtPieceFactory } from "@test/factories";
import { render, screen, waitFor } from "@testing-library/react";
import { api } from "@services/api";
import { mocked } from "ts-jest/utils";

import React from "react";

import { ArtCard } from "./art_card";

jest.mock("@services/api");
const mockApi = mocked(api, true);


describe("ArtCard", () => {
  let artPiece;

  beforeEach(() => {
    api.artPieces.image.mockResolvedValue({url:'https://theimage.whatever.com/thing.jpg'})
    artPiece = new ArtPiece(jsonApiArtPieceFactory.build());
  });

  it("matches the snapshot with all the props", async () => {
    const { container } = render(<ArtCard artPiece={artPiece} />);
    await waitFor(() => {
      screen.getByText("Sold")
    })
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot with minimal props", async () => {
    artPiece.dimensions = undefined;
    artPiece.price = undefined;
    const { container } = render(<ArtCard artPiece={artPiece} />);
    await waitFor(() => {
      screen.getByText("Sold")
    })
    expect(container).toMatchSnapshot();
  });
});
