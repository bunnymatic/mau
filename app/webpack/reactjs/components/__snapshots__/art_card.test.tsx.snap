// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtCard matches the snapshot with all the props 1`] = `
<div>
  <div
    class="art-card"
  >
    <div>
      <div
        class="image"
        style="background-image: url(original.png);"
      />
      <span
        class="art-card__sold"
      >
        Sold
      </span>
      <div
        class="art-card__description"
      >
        <div
          class="art-card__description-item"
        >
          10x 20
        </div>
        <div
          class="art-card__description-item"
        >
          <span
            class="art-card__price art-card__price--sold"
          >
            $123
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArtCard matches the snapshot with minimal props 1`] = `
<div>
  <div
    class="art-card"
  >
    <div>
      <div
        class="image"
        style="background-image: url(original.png);"
      />
      <span
        class="art-card__sold"
      >
        Sold
      </span>
      <div
        class="art-card__description"
      >
        <div
          class="art-card__description-item"
        >
          the title goes here
        </div>
      </div>
    </div>
  </div>
</div>
`;
